provisioner.go

```diff
@@ -322,75 +335,66 @@ func (p *LocalPathProvisioner) createHelperPod(action ActionType, cmdsForPath []
-	helperPod := &v1.Pod{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      string(action) + "-" + name,
-			Namespace: p.namespace,
-		},
-		Spec: v1.PodSpec{
-			RestartPolicy:      v1.RestartPolicyNever,
-			NodeName:           node,
-			ServiceAccountName: p.serviceAccountName,
-			Tolerations: []v1.Toleration{
-				{
-					Operator: v1.TolerationOpExists,
+	vols := []v1.Volume{
+		{
+			Name: "data",
+			VolumeSource: v1.VolumeSource{
+				HostPath: &v1.HostPathVolumeSource{
+					Path: parentDir,
+					Type: &hostPathType,
 				},
 			},
-			Containers: []v1.Container{
-				{
-					Name:    "local-path-" + string(action),
-					Image:   p.helperImage,
-					Command: append(cmdsForPath, filepath.Join("/data/", volumeDir)),
-					VolumeMounts: []v1.VolumeMount{
+		},
+		{
+			Name: "script",
+			VolumeSource: v1.VolumeSource{
+				ConfigMap: &v1.ConfigMapVolumeSource{
+					LocalObjectReference: v1.LocalObjectReference{
+						Name: p.configMapName,
+					},
+					Items: []v1.KeyToPath{
 						{
-							Name:      "data",
-							ReadOnly:  false,
-							MountPath: "/data/",
+							Key:  "setup",
+							Path: "setup",
 						},
 						{
-							Name:      "script",
-							ReadOnly:  false,
-							MountPath: "/script",
-						},
-					},
-					ImagePullPolicy: v1.PullIfNotPresent,
-				},
-			},
-			Volumes: []v1.Volume{
-				{
-					Name: "data",
-					VolumeSource: v1.VolumeSource{
-						HostPath: &v1.HostPathVolumeSource{
-							Path: parentDir,
-							Type: &hostPathType,
-						},
-					},
-				},
-				{
-					Name: "script",
-					VolumeSource: v1.VolumeSource{
-						ConfigMap: &v1.ConfigMapVolumeSource{
-							LocalObjectReference: v1.LocalObjectReference{
-								Name: p.configMapName,
-							},
-							Items: []v1.KeyToPath{
-								{
-									Key:  "setup",
-									Path: "setup",
-								},
-								{
-									Key:  "teardown",
-									Path: "teardown",
-								},
-							},
+							Key:  "teardown",
+							Path: "teardown",
 						},
 					},
 				},
 			},
 		},
 	}
+	volMnts := []v1.VolumeMount{
+		{
+			Name:      "data",
+			ReadOnly:  false,
+			MountPath: parentDir,
+		},
+		{
+			Name:      "script",
+			ReadOnly:  false,
+			MountPath: "/script",
+		},
+	}
+	tolerations := []v1.Toleration{
+		{
+			Operator: v1.TolerationOpExists,
+		},
+	}
+	helperPod := p.helperPod.DeepCopy()
+
+	helperPod.Namespace = p.namespace
+	helperPod.Spec.NodeName = node
+	helperPod.Spec.ServiceAccountName = p.serviceAccountName
+	helperPod.Spec.RestartPolicy = v1.RestartPolicyNever
+	helperPod.Spec.Tolerations = append(helperPod.Spec.Tolerations, tolerations...)
+	helperPod.Spec.Volumes = append(helperPod.Spec.Volumes, vols...)
+	helperPod.Spec.Containers[0].VolumeMounts = append(helperPod.Spec.Containers[0].VolumeMounts, volMnts...)
+	helperPod.Spec.Containers[0].Command = cmdsForPath
+	helperPod.Spec.Containers[0].Args = []string{filepath.Join(parentDir, volumeDir), strconv.FormatInt(sizeInBytes, 10), volumeMode}
```

Comment:
We should add a prefix for each parameter now, it's getting confusing.

Reply:
i add a prefix for these parameters. @yasker could you check for this?

Reply:
I meant, we can have the script accept parameters by name rather than just order, e.g.

`./setup --path <path> --size <size> --mode <mode>...`

Or we can just do 
`./setup -p <path> -s <size> -m <mode>...`

So the arguments are not solely decided by the order, make it easier to skip a few argument e.g. `size` if not needed, as well as make it clear to the caller how should it use the script.

Reply:
ok, good idea, it is done now.